fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/synonym/SynonymAnalyzer.java

```diff
@@ -96,19 +99,29 @@ public String getName() {
         @SuppressWarnings("deprecation")
         @Nonnull
         @Override
-        public Analyzer getIndexAnalyzer(@Nonnull Index index) {
-            return new StandardAnalyzer();
+        public Map<TextLanguage, Analyzer> getIndexAnalyzerMap(@Nonnull Index index) {
+            Map<TextLanguage, Analyzer> map = new HashMap<>();
+            for (TextLanguage language : TextLanguage.values()) {
+                map.put(language, new StandardAnalyzer());
+            }
+            return map;
         }
 
         @SuppressWarnings("deprecation")
         @Nonnull
         @Override
-        public Analyzer getQueryAnalyzer(@Nonnull Index index, @Nonnull Analyzer indexAnalyzer) {
+        public Map<TextLanguage, Analyzer> getQueryAnalyzerMap(@Nonnull Index index, @Nonnull Map<TextLanguage, Analyzer> indexAnalyzerMap) {
             String name = index.getOption(LuceneIndexOptions.TEXT_SYNONYM_SET_NAME_OPTION);
             if (name == null) {
                 name = EnglishSynonymMapConfig.CONFIG_NAME;
             }
-            return new SynonymAnalyzer(EnglishAnalyzer.ENGLISH_STOP_WORDS_SET, name);
+            final Analyzer analyzer = new SynonymAnalyzer(EnglishAnalyzer.ENGLISH_STOP_WORDS_SET, name);
```

Comment:
Is this an intermediate step? All the language calculation still leads to the same analyzer?

Reply:
added comments